api/server/router/container/container_routes.go

```diff
@@ -480,85 +480,98 @@ func (s *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 		return err
 	}
 
-	name := r.Form.Get("name")
-
 	config, hostConfig, networkingConfig, err := s.decoder.DecodeConfig(r.Body)
 	if err != nil {
 		return err
 	}
-	version := httputils.VersionFromContext(ctx)
-	adjustCPUShares := versions.LessThan(version, "1.19")
+	var (
+		name            = r.Form.Get("name")
+		version         = httputils.VersionFromContext(ctx)
+		adjustCPUShares bool
+	)
 
-	// When using API 1.24 and under, the client is responsible for removing the container
-	if hostConfig != nil && versions.LessThan(version, "1.25") {
-		hostConfig.AutoRemove = false
-	}
+	if hostConfig != nil {
+		if versions.LessThan(version, "1.19") {
+			adjustCPUShares = true
+		}
```

Comment:
Why change `adjustCPUShares` to be conditional on `hostConfig != nil` when it wasn't before?